app/services/location_suggestions.rb

```diff
@@ -0,0 +1,49 @@
+class LocationSuggestions
+  attr_reader :query, :cache, :cache_expiration
+
+  def initialize(query, cache: Rails.cache, cache_expiration: 7.days)
+    @query = query
+    @cache = cache
+    @cache_expiration = cache_expiration
+  end
+
+  def call
+    return [] if query.blank?
+
+    cached_suggestions || fetch_suggestions_and_cache
```

Comment:
Is there a reason why we're not using the Rails caching syntax here?
```suggestion
Rails.cache.fetch(cache_key, expires_in: cache_expiration) do
        fetch_suggestions
    end
```